ion-schema/src/result.rs

```diff
@@ -40,6 +40,28 @@ pub enum IonSchemaError {
     },
 }
 
+// io::Error does not implement PartialEq, which precludes us from simply deriving an implementation.
+impl PartialEq for IonSchemaError {
+    fn eq(&self, other: &Self) -> bool {
+        use IonSchemaError::*;
+        match (self, other) {
+            // We can compare the io::Errors' ErrorKinds, offering a weak definition of equality.
+            (IoError { source: s1 }, IoError { source: s2 }) => s1.kind() == s2.kind(),
+            (
+                UnresolvableSchemaError { description: s1 },
```

Comment:
I'm a little uneasy about having the equality function test only for description string equality, which can be pretty fragile. Developers are likely to write unit tests against a given error string, which means we can't change that error string down the road without also breaking their tests.

We do description string comparisons like this in `ion-rust`'s `IonError` type, but the description messages are intentionally terse and do not include lots of contextual information like position. Even there I don't love it.

I suppose ideally we'd have fields besides the description that characterized the nature of the failure. For example, a `schema_name` field in each `UnresolvableSchemaError`, or an enum of reasons that an `InvalidSchemaError` occurred. Well-defined, tightly-scoped fields that we're comfortable committing to using in our `PartialEq` impl.

This doesn't need to block this PR, but we should think it through before 1.0.

Reply:
I have created an issue for this: https://github.com/amazon-ion/ion-schema-rust/issues/141